src/main/java/com/epam/aidial/cfg/service/transfer/CoreConfigVersionAutoDetectService.java

```diff
@@ -0,0 +1,86 @@
+package com.epam.aidial.cfg.service.transfer;
+
+import com.epam.aidial.cfg.client.AnonymousCoreConfigClient;
+import com.epam.aidial.cfg.configuration.CoreConfigVersionProperties;
+import com.epam.aidial.cfg.configuration.logging.LogExecution;
+import com.google.common.cache.Cache;
+import com.google.common.cache.CacheBuilder;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Service;
+
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicReference;
+
+@Service
+@Slf4j
+@LogExecution
+public class CoreConfigVersionAutoDetectService {
+
+    private static final String CURRENT_VERSION_CACHE_KEY = "current-version";
+
+    private final AnonymousCoreConfigClient coreConfigClient;
+    private final CoreConfigVersionProperties coreConfigVersionProperties;
+    private final Cache<String, String> versionCache;
+    private final AtomicReference<String> nonExpiringCachedVersion;
+
+    public CoreConfigVersionAutoDetectService(AnonymousCoreConfigClient coreConfigClient, CoreConfigVersionProperties coreConfigVersionProperties) {
+        this.coreConfigClient = coreConfigClient;
+        this.coreConfigVersionProperties = coreConfigVersionProperties;
+        this.versionCache = CacheBuilder.newBuilder()
+                .expireAfterWrite(coreConfigVersionProperties.getCacheExpirationMs(), TimeUnit.MILLISECONDS)
+                .build();
+        this.nonExpiringCachedVersion = new AtomicReference<>();
+    }
+
+    /**
+     * Gets the Core version with retry mechanism.
+     * If all retries fail, falls back to the target version from properties.
+     * If auto-detection is disabled, returns the target version directly.
+     *
+     * @return the Core version
+     */
+    public String getVersion() {
+        if (!coreConfigVersionProperties.isAutoDetectEnabled()) {
+            String target = coreConfigVersionProperties.getTarget();
+            log.debug("Core version auto-detection is disabled. Using target version: {}", target);
+            return target;
+        }
+
+        String version = versionCache.getIfPresent(CURRENT_VERSION_CACHE_KEY);
+        if (version != null) {
+            log.debug("Using cached Core version: {}", version);
+            return version;
+        }
+
+        try {
+            version = getVersionFromCore();
+            versionCache.put(CURRENT_VERSION_CACHE_KEY, version);
+            nonExpiringCachedVersion.set(version);
+            return version;
+        } catch (Exception e) {
+            log.warn("Failed to get Core version. Reason: ", e);
+            return fallbackToTargetVersion();
```

Comment:
let's think if we really need this fallback. 
i think about case when admin app starts before admin app and see issues with such approach

Reply:
removed fallback logic to avoid inconsistencies between admin app and core